src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java

```diff
@@ -103,27 +107,30 @@ private void processIfIncludeEvidence(SscsCaseData caseData, PreSubmitCallbackRe
             List<DwpDocument> dwpDocuments = new ArrayList<>();
 
             for (AudioVideoEvidence audioVideoEvidence : caseData.getAudioVideoEvidence()) {
+                if (isSelectedEvidence(audioVideoEvidence, caseData)) {
```

Comment:
I think you can make this logic simple by fetching the selected file first and then apply the logic to add it into the right collection. You don't need to do that inside the loop as only one file is selected.